### DISCUSSION TITLE
GPU instead CPU segmentation

### DISCUSSION THREAD

**Discussion by lennart81** [ENHANCE-PET/MOOSE#106] [General]
Hello,
I just installed MooseZ and would like to run a few segmentations. I have an Nvidia GPU, however, the calculations are performed on my CPU. Can I force it to run on the GPU instead?

Thanks, Jan Lennart

**Comment by LalithShiyam** [comment]
Hi @lennart81,

I would be needing more information to help you. We automatically pick GPU if it's available. If it's not being done. There can be a lot of issues. 

Kindly let me know the following information: 

- OS: windows/Linux
- python version (python --version) 
- cuda version (nvidia-smi)

Cheers,
Lalith

**Reply by lennart81** [reply]
Hi Lalith, 
We have been in contact last week via LinkedIn. I wanted to give moose a try and it wasn't too hard for me (as a biologist) to get it started. However, it is running on my CPU only.
I am using Win10 pro, python 3.11.8 and cuda 12.4

any ideas?

Thanks, Jan

**Reply by Keyn34** [reply]
Hi @lennart81

For starters, try downgrading to Python 3.10 and give it another go. 
Let me know if this changes anything!

**Reply by lennart81** [reply]
hi @Keyn34 thanks for your quick reply. I deinstalled pyhton and reinstalled 3.10. however, same issue 100% CPU; 0% GPU

**Reply by LalithShiyam** [reply]
Hi @lennart81 , was the environment created with `python3.10 -m venv <your_env_name>`?

**Reply by lennart81** [reply]
@LalithShiyam yes, I removed the old environment and recreated a new one.

**Reply by LalithShiyam** [reply]
and even then the problem persists? @lennart81

**Reply by lennart81** [reply]
this is what I see. Maybe my NVIDIA GPU is not supported?!
![Clipboard_04-02-2024_01](https://github.com/ENHANCE-PET/MOOSE/assets/12213846/0eb48680-1876-41b0-aec9-552b91493598)

**Reply by lennart81** [reply]
it is a NVIDIA MX550

**Reply by LalithShiyam** [reply]
before installing moose: did you install PyTorch separately?

**Comment by lennart81** [comment]
I re-installed CUDA, NVIDIA drivers and moosez again yesterday. When I performed organ segmentation with moosez it took 14mins. 
maybe this is what I can expect from a mobile GPU on an office notebook? better than to wait >1h when running on a CPU

![image](https://github.com/ENHANCE-PET/MOOSE/assets/12213846/4678804b-4e95-4cab-b1a9-a79acda5e595)

**Reply by LalithShiyam** [reply]
Hi @lennart81, great. This is reasonable I guess. Would you be willing to add your 'error' and 'solution' to our FAQ.md, we are happy to add you as a collaborator :)
